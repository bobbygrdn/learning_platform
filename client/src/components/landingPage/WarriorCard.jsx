/* eslint-disable jsx-a11y/img-redundant-alt */
import React, { useEffect } from "react";
import { Image } from "react-bootstrap";

export default function WarriorCard({ warrior }) {
  const [warrior1, setWarrior1] = React.useState("");
  const [warrior2, setWarrior2] = React.useState("");
  const [isHovered, setIsHovered] = React.useState(false);

  useEffect(() => {
    switch (warrior) {
      case "Samurai":
        setWarrior1(process.env.PUBLIC_URL + "/resources/male_meijin.jpg");
        setWarrior2(process.env.PUBLIC_URL + "/resources/female_meijin.jpg");
        break;
      case "Legionnaire":
        setWarrior1(process.env.PUBLIC_URL + "/resources/female_imperator.jpg");
        setWarrior2(process.env.PUBLIC_URL + "/resources/male_imperator.jpg");
        break;
      case "Spartan":
        setWarrior1(process.env.PUBLIC_URL + "/resources/male_king.jpg");
        setWarrior2(process.env.PUBLIC_URL + "/resources/female_king.jpg");
        break;
      default:
        setWarrior1(
          process.env.PUBLIC_URL + "/resources/female_vikingking.jpg"
        );
        setWarrior2(process.env.PUBLIC_URL + "/resources/male_vikingking.jpg");
        break;
    }
  }, [warrior]);

  const handleMouseEnter = () => {
    setIsHovered(true);
  };

  const handleMouseLeave = () => {
    setIsHovered(false);
  };

  return (
    <div className="warriorCard">
      <h1 className="warriorTitle">{warrior}</h1>
      <Image
        className="card"
        src={isHovered ? warrior1 : warrior2}
        onMouseEnter={handleMouseEnter}
        onMouseLeave={handleMouseLeave}
        data-testid="card"
        alt="warrior image"
      />
    </div>
  );
}
