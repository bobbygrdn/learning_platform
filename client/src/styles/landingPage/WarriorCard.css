.warriorCards {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  margin-bottom: 2em;
}

.warriorList {
  display: flex;
  align-items: flex-start;
  list-style: none;
  pointer-events: none;
}

.warriorTitle {
  font-size: 2vw;
  text-align: center;
  margin-bottom: .5em;
  font-weight: normal;
}

.warriors {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
}

.card {
  width: 16vw;
  border: 2px solid #677d8c;
  box-shadow: 0px 0px 20px #677d8c;
  border-radius: 5%;
  margin-top: 0;
  margin-left: 0.5em;
  margin-right: 0.5em;
}

.card:hover {
  cursor: pointer;
}

@media screen and (max-width: 600px) {

  .card {
    width: 20vw;
  }

  .warriorName {
    padding-left: 1.5em;
    padding-right: 0.5em;
  }

  .warriorList :nth-child(1) {
    padding-left: 1.2em;
  }

  .warriorList :nth-child(2) {
    padding-left: 0.8em;
  }

  .warriorList :nth-child(3) {
    padding-left: 1em;
  }

  .warriorList :nth-child(4) {
    padding-left: 2.2em;
  }

  .warriorTitle {
    font-size: 4vw;
    text-align: center;
  }
}

@media screen and (min-width: 1000px) {
  .warriorName {
    font-size: 24px;
  }

  .warriorList :nth-child(2) {
    padding-left: 3em;
  }

  .warriorList :nth-child(3) {
    padding-left: 3em;
  }

  .warriorList :nth-child(4) {
    padding-left: 4em;
  }
}