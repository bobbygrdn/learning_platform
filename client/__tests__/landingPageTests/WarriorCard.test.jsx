import React from 'react';
import { render, screen } from '@testing-library/react';
import WarriorCard from '../../src/components/landingPage/WarriorCard';

it('renders without crashing', () => {
    const { container } = render(<WarriorCard />);
    expect(container).toMatchSnapshot();
})

it('should render card based on warrior type', () => {
    render(<WarriorCard />);
    expect(screen.getAllByTestId('card')).toHaveLength(1);
});
