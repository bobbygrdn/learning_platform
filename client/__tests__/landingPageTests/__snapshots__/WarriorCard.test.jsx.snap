// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <img
    alt="warrior image"
    class="card"
    data-testid="card"
    src="undefined/resources/male_king_viking.jpg"
  />
</div>
`;
